File: 프로그래머스/1/42748. K번째수/K번째수.java
import java.util.Arrays;

class Solution {
    public int[] solution(int[] array, int[][] commands) {
        int[] results = new int[commands.length];
        for (int i = 0; i < commands.length; i++) {
            int[] command = commands[i];
            int startIndex = command[0] - 1;
            int endIndex = command[1] - 1;
            int pickedIndex = command[2] - 1;
            int[] slicedArray = Arrays.copyOfRange(array, startIndex, endIndex + 1);
            Arrays.sort(slicedArray);
            results[i] = slicedArray[pickedIndex];
        }
        return results;
    }
}

File: 프로그래머스/1/42748. K번째수/K번째수.kt
class Solution {
    fun solution(array: IntArray, commands: Array<IntArray>): IntArray {
        val results = IntArray(commands.size)
        commands.forEachIndexed { i, command ->
            val (start, end, ordinal) = command
            val result = array.sliceArray(start - 1 until end).sorted()[ordinal - 1]
            results[i] = result
        }
        return results
    }
}